### PR TITLE
Encryption: Fix multiple data keys migration

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:
https://github.com/grafana/grafana/pull/47765 introduced an implicit database schema breaking change by renaming PK `name` column to `id`. 
This causes issues in HA setups in cases when the database schema has already been migrated to v9.x, but one of the instances runs v8.x code and can't insert a data key into the database due to scheme being different in v9.x.
It also makes impossible to rollback to previous Grafana versions without manually changing the db schema.

The goal of this PR is to fix migration, but preserve the ability to have multiple DEKs per day: as before, the DEK identifier is stored in the `name` column and it's used to decrypt secrets, but the current DEK lookup used for encrypting new secrets happens by `label` column. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

 _This PR makes the version compatible with v8.5.x, but incompatible with v9-beta1/beta2 (EE in beta versions relies on `id` being present in `data_keys` table). So there may be failures in HA setups when the db schema has been migrated to v9-beta3, but some of the instances keep running v9-beta1/beta2._

Requires https://github.com/grafana/grafana/pull/49740 to be merged first

### CODE REVIEW

**Comment by joanlopez** [comment]
JFYI and further context (cc/ @dafydd-t @bergquist @stephaniehingtgen  @hairyhenderson):

This pull request represents (another, implicit) breaking change (in terms of database migrations & code) respect to `v9.0.0-beta1`, but it's intended because we want to "revert" the breaking change introduced there and make `v9.0.0-beta3` (or directly `v9.0.0-stable`) backwards compatible with `v8.5.x`.

If we got it correctly, that's we agreed while working some of the recent incidents, but would be good to get your double-check and confirmation. Also pinging for awareness because it may/will cause another peak of temporary errors.

**Comment by bergquist** [comment]
> It also makes impossible to rollback to previous Grafana versions without manually changing the db schema.

I'm not sure we can do that. How can guarantee that 9.0 doesn't have bugs that wont force us to rollback?

**Comment by undef1nd** [comment]
>Can we upgrade from 9.0.0-beta1 to 9.0.0-beta3 and then back to 8.5.x? 

Yes, we can.
When upgrading from 9.0.0-beta1 to 9.0.0-beta3, there will be failures till all pods are on 9.0.0-beta3. And then we will be able to downgrade from 9.0.0-beta3 to 8.5.x.
@joanlopez please, correct me if I'm missing smth.

**Review by bergquist — APPROVED** [review]
> When upgrading from 9.0.0-beta1 to 9.0.0-beta3, there will be failures till all pods are on 9.0.0-beta3. And then we will be able to downgrade from 9.0.0-beta3 to 8.5.x.

Perfect! thank you :)

File: pkg/services/secrets/types.go [hunk 1]
```diff
@@ -9,8 +9,8 @@ var ErrDataKeyNotFound = errors.New("data key not found")
 
 type DataKey struct {
 	Active        bool
-	Id            string
-	Name          string
+	Id            string `xorm:"name"`
```
[the remark this answers is not in the document]
Comment (joanlopez) [reply pkg/services/secrets/types.go]: Good idea, yeah! Thanks!

**Review by joanlopez — APPROVED** [review]
Looks fine and additional manual tests seems to be working as expected, thanks!!

**Pull request merged**